index.js

```diff
@@ -0,0 +1,36 @@
+const TileServer = require("./server");
+
+const stopsQuery = `
+    SELECT ST_AsMVT('stops', 4096, 'geom', rows)
+    FROM (
+        SELECT
+            stop_id,
+            short_id,
+            name_fi,
+            name_se,
+            jore.stop_modes(stop.*, $5) AS type,
+            ST_AsMVTGeom(ST_Transform(point, 3857), ST_MakeEnvelope($1, $2, $3, $4, 3857), 4096, 0, false) AS geom
+        FROM jore.stop stop
+        WHERE ST_Transform(point, 3857) && ST_MakeEnvelope($1, $2, $3, $4, 3857)
+    ) AS rows`;
+
+const routesQuery = `
+    SELECT ST_AsMVT('routes', 4096, 'geom', rows)
+    FROM (
+        SELECT
+            route_id,
+            direction,
+            date_begin,
+            date_end,
+            mode,
+            ST_AsMVTGeom(ST_Transform(geom, 3857), ST_MakeEnvelope($1, $2, $3, $4, 3857), 4096, 0, true) AS geom
+        FROM jore.geometry geometry
+        WHERE $5 between date_begin and date_end and ST_Intersects(ST_Transform(geom, 3857), ST_MakeEnvelope($1, $2, $3, $4, 3857))
```

Comment:
How about using `ST_Transform(geom, 3857) && ST_MakeEnvelope($1, $2, $3, $4, 3857)` instead of `ST_Intersects(ST_Transform(geom, 3857), ST_MakeEnvelope($1, $2, $3, $4, 3857))` so that it uses the geometric index for the calculation?

Reply:
or actually just `geom && ST_MakeEnvelope($1, $2, $3, $4, 3857)` or changing the bbox to EPSG:4326 if the coordinate systems need to match

Reply:
Or we could add the following index for the table
`CREATE INDEX route_wgs_merator_idx ON jore.geometry USING GIST ( ST_Transform(geom, 3857) );`

Reply:
ST_Intersects is required to only include linestrings that really intersect with the tile bounds. Removing ST_Transform seemed to improve performance quite a bit.